pydruid/utils/filters.py

```diff
@@ -20,45 +20,59 @@
 
 
 class Filter:
-    def __init__(self, **args):
-
-        if 'type' not in args.keys():
-            self.filter = {"filter": {"type": "selector",
-                                      "dimension": args["dimension"],
-                                      "value": args["value"]}}
-
-        elif args["type"] == "javascript":
-            self.filter = {"filter": {"type": "javascript",
-                                      "dimension": args["dimension"],
-                                      "function": args["function"]}}
-        elif args["type"] == "and":
-            self.filter = {"filter": {"type": "and",
-                                      "fields": args["fields"]}}
-        elif args["type"] == "or":
-            self.filter = {"filter": {"type": "or",
-                                      "fields": args["fields"]}}
-        elif args["type"] == "not":
-            self.filter = {"filter": {"type": "not",
-                                      "field": args["field"]}}
-        elif args["type"] == "in":
-            self.filter = {"filter": {"type": "in",
-                                      "dimension": args["dimension"],
-                                      "values": args["values"]}}
-        elif args["type"] == "regex":
-            self.filter = {"filter": {"type": "regex",
-                                      "dimension": args["dimension"],
-                                      "pattern": args["pattern"]}}
-        elif args["type"] == "bound":
-            self.filter = {"filter": {"type": "bound",
-                                      "dimension": args["dimension"],
-                                      "lower": args["lower"],
-                                      "lowerStrict": args["lowerStrict"],
-                                      "upper": args["upper"],
-                                      "upperStrict": args["upperStrict"],
-                                      "alphaNumeric": args["alphaNumeric"]}}
+
+    # filter types supporting extraction function
+    _FILTERS_WITH_EXTR_FN = ('selector', 'regex', 'javascript', 'in', 'bound',
+                             'extraction')
+
+    def __init__(self, extraction_function=None, **args):
```

Comment:
Should the extraction_function argument be added to the constructor of the child filters like Bound, Dimension etc?